### PR TITLE
LL-2720 LL-2729 LL-2731 (Algo): add base UI

### PR DESCRIPTION
**Pull request by LFBarreto**
(Algo): add base account UI + asa using mocks

HODL: needs live-common bump that includes algo update.

### Type

Feature

### Context

LL-2720 L-2729 LL-2731

### Parts of the app affected / Test plan

Start the app using mocks with the correct live-common linked to it and experimental currencies toggled on for algorand.

### CODE REVIEW

File: src/families/algorand/OptInFlow/01-SelectToken.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  onPress,
+  onDisabledPress,
+  disabled,
+}: {
+  item: TokenCurrency,
+  onPress: () => void,
+  onDisabledPress: () => void,
+  disabled: boolean,
+}) => {
+  const tokenId = extractTokenId(item.id);
```
Comment (gre) [inline src/families/algorand/OptInFlow/01-SelectToken.js:45]: mmh this is really weird we have to do this to me. also why are we displaying the "id" on the UI? why is not not part in the name in the first place?

File: src/families/algorand/OptInFlow/01-SelectToken.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+};
+
+export default function DelegationStarted({ navigation, route }: Props) {
+  const { account } = useSelector(accountScreenSelector(route));
+  const { t } = useTranslation();
+
+  invariant(account, "Account required");
+
+  const mainAccount = getMainAccount(account, undefined);
+  const bridge = getAccountBridge(mainAccount, undefined);
```
Comment (gre) [inline src/families/algorand/OptInFlow/01-SelectToken.js:92]: second parameter can be omitted

File: src/families/algorand/OptInFlow/01-SelectToken.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+  const { transaction } = useBridgeTransaction(() => {
+    const t = bridge.createTransaction(mainAccount);
+
+    return {
+      account,
+      transaction: bridge.updateTransaction(t, {
+        mode: "optIn",
+        assetId: null,
+        /** @TODO remove this once the bridge handles it */
```
Comment (gre) [inline src/families/algorand/OptInFlow/01-SelectToken.js:107]: (similar question as on LLD)

File: src/families/algorand/OptInFlow/03-Validation.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+export default function Validation({ navigation, route }: Props) {
+  const { account } = useSelector(accountScreenSelector(route));
+  invariant(account, "account is required");
+  const dispatch = useDispatch();
+
+  const [signing, signed] = useSignWithDevice({
+    context: "AlgorandOptIn",
+    account,
+    parentAccount: undefined,
```
Comment (gre) [inline src/families/algorand/OptInFlow/03-Validation.js:45]: shouldn't be required 🤔

File: src/families/algorand/Rewards/index.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+import CurrencyUnitValue from "../../../components/CurrencyUnitValue";
+import CounterValue from "../../../components/CounterValue";
+import { ScreenName, NavigatorName } from "../../../const";
+
+type Props = {
+  account: Account,
+};
+
+const RewardsSection = ({ account }: Props) => {
+  const { rewards } = account.algorandResources || {};
```
Comment (gre) [inline src/families/algorand/Rewards/index.js:27]: this shouldn't be needed. do an invariant to not have bad flowtype inference (rewards must be a weird type here?)

File: src/families/algorand/SubAccountList.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    accountScreenSelector({ params: { accountId } }),
+  );
+  const onReceiveClick = useCallback(() => {
+    navigation.navigate(NavigatorName.AlgorandOptInFlow, {
+      screen: ScreenName.AlgorandOptInSelectToken,
+      params: { accountId },
+    });
+  }, [navigation, accountId]);
+  const howAsaWorks = useCallback(() => {
+    /** @TODO replace this url */
```
Comment (gre) [inline src/families/algorand/SubAccountList.js:50]: remove the comment? the code here is fine.

File: src/families/cosmos/SendRowsFee.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  account: AccountLike,
+  transaction: Transaction,
+};
+
+export default function CosmosFeeRow({ account, transaction }: Props) {
+  const extraInfoFees = useCallback(() => {
+    Linking.openURL(urls.feesMoreInfo);
+  }, []);
+
+  if (transaction.family !== "cosmos") return null;
```
Comment (gre) [inline src/families/cosmos/SendRowsFee.js:29]: shouldn't this be an invariant?
also why is there cosmos in the PR of algo?

File: src/families/algorand/OptInFlow/01-SelectToken.js (hunk 1 above)
Comment (LFBarreto) [reply src/families/algorand/OptInFlow/01-SelectToken.js:45]: we show the name and the Asset id it was in the designs

File: src/families/cosmos/SendRowsFee.js (hunk 7 above)
Comment (LFBarreto) [reply src/families/cosmos/SendRowsFee.js:29]: I figured while integrating Algo that the fees were missing in the send summary for Cosmos

File: src/families/algorand/OptInFlow/03-Validation.js (hunk 4 above)
Comment (LFBarreto) [reply src/families/algorand/OptInFlow/03-Validation.js:45]: flow requires it :sweat_smile:

**Pull request merged**